beacon-chain/rpc/eth/events/events.go

```diff
@@ -219,7 +220,23 @@ func (s *Server) handleStateEvents(
 			return streamData(stream, HeadTopic, head)
 		}
 		if _, ok := requestedTopics[PayloadAttributesTopic]; ok {
-			if err := s.streamPayloadAttributes(stream); err != nil {
+			head, ok := event.Data.(*ethpb.EventHead)
+			if !ok {
+				return nil
+			}
+			if err := s.streamPayloadAttributes(stream, head.Slot); err != nil {
+				log.WithError(err).Error("Unable to obtain stream payload attributes")
+			}
+			return nil
+		}
+		return nil
+	case statefeed.MissedSlot:
+		if _, ok := requestedTopics[PayloadAttributesTopic]; ok {
+			emitSlot, ok := event.Data.(primitives.Slot)
+			if !ok {
+				return nil
+			}
+			if err := s.streamPayloadAttributes(stream, emitSlot); err != nil {
```

Comment:
Nice! another option is to do `s.CurrentSlot()` from one of the fetcher. This is fine too